zulip_bots/zulip_bots/bots/calendarbot/calendarbot.py

```diff
@@ -1,154 +1,250 @@
-        
 
     @dataclass
     class MeetingDetails:
-        sender_id: int
-        invitees: list
+        name: str
+        location: str
+        description: str
         meeting_start: datetime
         meeting_end: datetime
-        name: str
-        summary: str
         length_minutes: int
+        sender_email: str
+        invitees: list
+
+
+    def initialize(self, bot_handler: BotHandler) -> None:
+        storage = bot_handler.storage
 
+        # Add key values in the bot_handler storage
+        if not storage.contains("confirmation"):
+            storage.put("confirmation", None)
+        if not storage.contains("starttime"):
+            storage.put("starttime", None)
+        if not storage.contains("duration"):
+            storage.put("duration", None)
 
-    def usage(self):
-        return """Creates calendar meeting for all participants on chat."""
 
-    def handle_message(self, message:dict, bot_handler):
-        self.message = message
-        self.bot_handler = bot_handler
-        message_content = message['content'].lower()
+    def usage(self) -> str:
+        return (
+            "Calendar Bot creates a calendar meeting for all participants on chat. "
+            "Enter a Zulip global time `<time` and then a duration in minutes (e.g. 60 for 1 hour). "
+            "Default event duration is 30 mins and default email addresses are from user accounts. "
+        )
         
-        #instantiating Calendar() here for now due to bug when trying to move it to init, works but may cause issues in future
+
+    def handle_message(self, message: Dict[str, Any], bot_handler: BotHandler) -> None:
+        storage = bot_handler.storage
+        confirmation = storage.get("confirmation")
+        starttime = storage.get("starttime")
+        duration = storage.get("duration")
+
+        # TODO: Instantiating Calendar here for now due to bug when trying to move it to init, works but may cause issues in future
         cal = Calendar()
 
-        
-        #casematch is overkill for now, but i suspect future cases to be added
-        match message_content.split():
-            case [datetime_input,length_minutes]:         
-                logging.debug(f"Datetime_input:{datetime_input},length_minutes {length_minutes}")
-                self.create_and_send_meeting(cal, datetime_input, length_minutes)
-            case [datetime_input]: 
-                if datetime_input == 'auth':
-                    authenticate_google()
-                logging.debug(f"Datetime_input:{datetime_input}")
-                self.create_and_send_meeting(cal, datetime_input)
-
-            case _:
-                self.input_error_reply()
-
-    def create_and_send_meeting(self, cal, datetime_input, length_minutes=30):
+        # Parse the initial message containing the event time and duration info
+        if confirmation is None and starttime is None:
+            bot_response = self.parse_first_message(message, bot_handler)
```

Comment:
Nit: Is it always the first message? Method name is somewhat misleading - maybe something like "parse_meeting_create_message" and "parse_confirmation_message")